tests/users/test_api_resources.py

```diff
@@ -49,6 +49,23 @@ def test_user_registration(self, send_email_function):
                 self.assertFalse(user.is_admin)
                 self.assertFalse(user.is_email_verified)
 
+    @patch('flask_mail._MailMixin.send', side_effect=mail_send_mocked)
+    def test_user_registration_with_optional_fields(self, send_email_function):
```

Comment:
If you're testing if the API can work with those fields as optional, it makes sense not sending them in the request, right?
You can have 2 tests:
1. Sending these fields and seeing if these fields are set correctly (try set these as false as well)
2. Without sending these fields and see if the default is false

Reply:
You can also test, just sending one fo the fields and if the values are set correctly, i.e., only the value sent will be changed, and the validation will pass because the other field is optional.

Reply:
Thank you @isabelcosta, I will add tests for the different use cases.

Reply:
Thank you @flacode !